include/deal.II/particles/generators.h

```diff
@@ -63,6 +72,99 @@ namespace Particles
       const Mapping<dim, spacedim> &      mapping =
         StaticMappingQ1<dim, spacedim>::mapping);
 
+    /**
+     * A function that generates one particle at a random location in cell @p cell and with
+     * index @p id. The function expects a random number generator to avoid the expensive generation
+     * and destruction of a generator for every particle and optionally takes
+     * into account a mapping for the cell. The algorithm implemented in the
+     * function is described in
+     *
+     * Gassmoeller, R., Lokavarapu, H., Heien, E., Puckett, E. G., & Bangerth,
+     * W. (2018). Flexible and Scalable Particle‐in‐Cell Methods With Adaptive
+     * Mesh Refinement for Geodynamic Computations. Geochemistry, Geophysics,
+     * Geosystems, 19(9), 3596-3604.
```

Comment:
You can now put this into `doc/doxygen/references.bib` and reference it with `@cite GLHPB18`.